common/usermode.c

```diff
@@ -124,13 +124,43 @@ static inline long __user_text syscall(long syscall_nr, long arg1, long arg2, lo
     return return_code;
 }
 
+inline long __user_text sysenter(long syscall_nr, long arg1, long arg2, long arg3,
+                                        long arg4, long arg5) {
+    register long return_code asm(STR(_ASM_AX));
+    register long _arg4 asm("r8") = arg4;
+    register long _arg5 asm("r9") = arg5;
+    register long _arg3 asm("r10") = arg3;
```

Comment:
Is it an issue that these can't be used in legacy mode, i.e. in 32 bit mode?

Reply:
It will be one day... let's maybe add a BUILD_BUG_ON() assert here for 32bit builds?

Reply:
Not needed, I guess, as the compiler will already complain that it knows nothing about these registers ;)